### PR TITLE
conformance check rebase v419

### PR DESCRIPTION
**Pull request by xinyunliu**
ported 6 patches from v4.14 to v4.19

### CODE REVIEW

File: drivers/gpu/drm/i915/gvt/gvt.h [hunk 1]
```diff
@@ -700,7 +702,7 @@ static inline bool intel_gvt_mmio_has_mode_mask(
  *
  */
 static inline bool intel_gvt_mmio_is_in_ctx(
-			struct intel_gvt *gvt, unsigned int offset)
+		struct intel_gvt *gvt, unsigned int offset)
```
Comment (minhe1) [inline drivers/gpu/drm/i915/gvt/gvt.h]: unnecessary change.

File: drivers/gpu/drm/i915/gvt/gvt.h [hunk 2]
```diff
@@ -712,11 +714,36 @@ static inline bool intel_gvt_mmio_is_in_ctx(
  *
  */
 static inline void intel_gvt_mmio_set_in_ctx(
-			struct intel_gvt *gvt, unsigned int offset)
+		struct intel_gvt *gvt, unsigned int offset)
```
Comment (minhe1) [inline drivers/gpu/drm/i915/gvt/gvt.h]: ditto

File: drivers/gpu/drm/i915/gvt/mmio.c [hunk 3]
```diff
@@ -213,6 +213,15 @@ int intel_vgpu_emulate_mmio_write(struct intel_vgpu *vgpu, uint64_t pa,
 	if (ret < 0)
 		goto err;
 
+	if (vgpu->entire_nonctxmmio_checked
+		&& intel_gvt_mmio_is_non_context(vgpu->gvt, offset)
+		&& vgpu_vreg(vgpu, offset)
+			!= *(u32 *)(vgpu->gvt->mmio.mmio_host_cache + offset)) {
```
Comment (minhe1) [inline drivers/gpu/drm/i915/gvt/mmio.c]: can we move gvt_host_reg to a common header file so that we can use it here?

File: drivers/gpu/drm/i915/gvt/gvt.h (hunk 1 above)
Comment (xinyunliu) [reply drivers/gpu/drm/i915/gvt/gvt.h]: Will correct it.

File: drivers/gpu/drm/i915/gvt/mmio.c (hunk 3 above)
Comment (xinyunliu) [reply drivers/gpu/drm/i915/gvt/mmio.c]: Sure. I planed to port my patch for 4.14 after it got  reviewed and merged.
I can combine them too. How do you think?

Comment (minhe1) [reply drivers/gpu/drm/i915/gvt/mmio.c]: I would like to combine them together to reduce the patch numbers, but it's up to you.

**Comment by xinyunliu** [comment]
close it. merged in 419 rebase.
